public/js/controllers/mempool.js

```diff
@@ -61,6 +61,47 @@
         $(newRowHtml).insertBefore(rows.first())
     }
 
+    function filterVotesLastBlock(target) {
+        var best_block_hash=$(target).data("hash");
+        var best_block_height=$(target).text();
+        console.log(best_block_height);
+        $('*[data-target="mempool.voteTransactions"] tr').each(function(i,el) {
+            var vote_validation_hash=$(this).data("blockhash");
+            var vote_block_height=$(this).data("height");
+            if(vote_block_height>best_block_height) {
+                $(this).closest("tr").addClass("upcoming-vote");
+                $(this).closest("tr").removeClass("old-vote");
+            }else if(vote_validation_hash!=best_block_hash){
+                $(this).closest("tr").addClass("old-vote");
+                $(this).closest("tr").removeClass("upcoming-vote");
+                $(this).find("td.last_block").text("Invalid");
+            }else{
+                $(this).closest("tr").removeClass("old-vote");
+                $(this).closest("tr").removeClass("upcoming-vote");
+                $(this).find("td.last_block").text("Valid");  
+            }
+        })
+    };
+
+    function sortVotesTable() {
+        var $table = $('*[data-target="mempool.voteTransactions"] tr').parent();
+        var $rows =$('*[data-target="mempool.voteTransactions"] tr');
+        $rows.sort(function(a, b){
+            var heightA = parseInt($('td:nth-child(2)',a).text());
+            var heightB = parseInt($('td:nth-child(2)',b).text());
+            if(heightA == heightB) {
+                 var indexA = parseInt($('td:nth-child(3)',a).text());
+                 var indexB = parseInt($('td:nth-child(3)',b).text());
+                 return (indexA - indexB);
+            }else{
+                return (heightB - heightA);
+            }
+        });
+        $.each($rows, function(index, row){
```

Comment:
Instead of looping, it's more efficient to update the html in a single operation via `$table.html($rows);`